CSharp/Library/Microsoft.Bot.Builder/Dialogs/PromptDialog.cs

```diff
@@ -637,7 +637,7 @@ protected override bool TryParse(IMessageActivity message, out double result)
         [Serializable]
         public class PromptChoice<T> : Prompt<T, T>
         {
-            private IDictionary<T, IEnumerable<T>> choices;
+            private IDictionary<T, IEnumerable<T>> synonyms;
```

Comment:
@msft-shahins, we are not fully convinced with this. We are thinking that probably the synonyms could be part of the PromptOptions and instead of being a Dictionary, this could be plain list as the actual options and descriptions. In that way, they will follow kind of the same pattern (question aside, should having the same or more descriptions than options be enforced? I think that right know you could run in a situation where you can have less descriptions than options and I think the method that build the card will fail.